City-of-Helsinki/drupal-helfi-rekry
# UHF-5920: Listing language

# [UHF-5920](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5920)
Add capability to filter job listings by language.

## What was done
* Add new filter for language options
  * Filtering displays only nodes with selection as their original language
* Change search logic to use IDs rather than taxonomy names for filtering
* Make required changes to index settings

## How to install

Set up your index: 
* Run make ps. Note the port that your drupal-helfi-rekry-elastic container is running in.
* Add :point_up: to your .env as the ELASTIC_PROXY_URL variable (http:localhost:{port-elastic-runs-in})
* Run make up
* make shell into the container. Verify that your ELASTIC_PROXY_URL is set by running printenv | grep ELASTIC_PROXY_URL
* Run `drush cr; drush sapi-r; drush sapi-i`

Make sure your instance is up and running on correct branch.
  * `git fetch; git checkout origin/UHF-5920-listing-language`
  * `make fresh` OR `make drush-cr; make drush-cim;`
* Run `make drush-cr`

## How to test
 * Add a `job_search` paragraph to a landing page **without** value in `field_job_search_result_page`
 * Try out the language filter
   * Making a selection should display nodes that have the selection as the original language (this relies on data from integration, so these might not he 100% accurate)
   * Since we automatically "translate" nodes to have all language versions (to make swedish listings more visible) don't be thrown off by aggregated values (in parentheses) not adding up

## Designers review
